src/datatypes/src/types/cast.rs

```diff
@@ -16,6 +16,21 @@ use crate::data_type::{ConcreteDataType, DataType};
 use crate::error::{self, Error, Result};
 use crate::types::TimeType;
 use crate::value::Value;
+use crate::vectors::Helper;
+
+/// Used to cast the value to dest ConcreteDataType temporarily.
+/// To keep the same behavior as arrow-rs.
+pub fn cast(src_value: Value, dest_type: &ConcreteDataType) -> Result<Value> {
```

Comment:
Looks like we don't need this function, already have `ConcreteDataType#try_cast(Value)`

https://greptimedb.rs/datatypes/data_type/trait.DataType.html#tymethod.try_cast

Reply:
The cast for decimal is complex, it should impl `cast string from/to decimal`, `cast numeric from/to decimal`, and `decimal from/to decimal with diff precision and scale`,  which is about 1000 rows, for this pr, it's too big. 
In addition, there are still many imperfections in `ConcreteDataType#try_cast` that I implement currently, and based on arrow::cast may be a better choice.

Reply:
I got u.